### PR TITLE
Disable swap form when Gnosis safe in readonly mode

### LINKED ISSUE #311: Gnosis Safe App issue with READ ONLY
**Description**

Issue with Gnosis Safe App when you are in read-only safe. It allows you to create orders, when it shouldn't

See:
https://github.com/gnosis/cowswap/pull/2495#issuecomment-1050774896

> Original issue https://github.com/gnosis/cowswap/issues/2499 by @anxolin

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #311

When Cow DAPP is opened inside Gnosis safe in Read-only mode, then we don't allow user to swap and show disabled button with text "Read Only"

image

  # To Test

https://github.com/cowprotocol/cowswap/issues/311#issuecomment-1201053615

  # Background

I had to update `@gnosis.pm/safe-apps-web3-react` package up to `1.5.0` version because the old version doesn't have full typings of SafeInfo (particularly `isReadOnly` flag)

### CODE REVIEW

**Comment by shoom3301** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by shoom3301** [comment]
@elena-zh please, check swap when user is not in read-only mode

File: src/custom/hooks/useGnosisSafeInfo.ts
Comment (alfetopito) [inline src/custom/hooks/useGnosisSafeInfo.ts:16]: Minor nitpick:

```suggestion
      gnosisSafe.getSafeInfo().then(setGnosisSafeInfo)
```

**Comment by elena-zh** [comment]
Hey @shoom3301 , changes LGTM when I open the Cowswap app in the Safe, and try to do a trade there when I'm on read-only mode. 

However, still I can see 'Swap' button enabled when I connect to the Safe using WC. See the video:  https://watch.screencastify.com/v/bb2CQ1Z8E7Q8jMeZ1oGo

Then, I can also press on the Swap button when I connected to the Safe mobile app in IOS, and the safe is in read-only mode. 

I mentioned all these cases in the 'Use cases' section in the https://github.com/cowprotocol/cowswap/issues/311#issuecomment-1201053615

Besides, according to the 2nd use-case, I propose to disable not only 'Swap' button, but Wrap/Unwrap/Approve tokens buttons on the Swap form, and Convert/Claim buttons on the Account page. @alfetopito , WDYT? Or I should create a separate task for this? 

Thanks!

**Review by alfetopito — APPROVED** [review]
I have one minor nitpick on the code.
Loved the unittests!

When loading it as an app, works great.

I mentioned we might want to disable approvals/etc, but while testing I don't think it's necessary. It's pretty clear it's not allowed when trying to trigger a tx from within the app
Screen Shot 2022-08-02 at 09 48 43

Now, it's still possible to place orders when not a Safe owner when connected via WalletConnect.
Not sure there is anything we can do about that though, given that we get no info as to which wallet is the connected owner in this case.

**Comment by elena-zh** [comment]
According to the discussion in the https://cowservices.slack.com/archives/C0361CDG8GP/p1659423106883899 , it is impossible to do anything regarding wallet connect. The same is with the iOS safe mobile app. 
Then, there is no need to implement the same logic for Wrap/Unwrap/Approve tokens buttons on the Swap form, and Convert/Claim buttons on the Account page. So the current changes are approved.

**Pull request merged**
